### PR TITLE
CONSOLE-2758: Bump webpack-cli and webpack-dev-server to latest version

### PR DESCRIPTION
**Pull request by vojtechszocs**
- [x] `webpack-cli` bumped from `3.3.9` to `4.5.0`
- [x] `webpack-dev-server` bumped from `3.11.0` to `3.11.1`
- [x] add matching `@types/webpack-dev-server` dev-dependency

The `dev` script has been updated to invoke `webpack-dev-server` via webpack CLI.

Starting with `webpack-cli` v4, this is the standard way to run the dev-server. See [here](https://github.com/webpack/webpack-dev-server/issues/2759#issuecomment-706753844) for details.

```sh
webpack serve --mode=development --progress
```

The `analyze` script has been updated to match changes in `webpack-cli` v4. See [here](https://webpack.js.org/api/cli/#progress) for details.

Our main `yarn.lock` file contains some resolutions pointing to `https://registry.npmjs.org/` instead of the expected `https://registry.yarnpkg.com/` and found [this Yarn issue](https://github.com/yarnpkg/yarn/issues/3330) which is fixed in Yarn v2 only.

### CODE REVIEW

**Comment by vojtechszocs** [comment]
cc @spadgett @christianvogt

**Comment by vojtechszocs** [comment]
Right now, latest `webpack-dev-server` is `3.11.2` and latest `@types/webpack-dev-server` is `3.11.1` - I've opted to bump both to `3.11.1` for sake of consistency.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
Since this is a dev-only dependency, this does not need QE review.

**Review by zherman0 — APPROVED** [review]
Looks good, will complete the other bump PRs once this merges

**Comment by spadgett** [comment]
Holding until #8289 merges to unblock CI.

/hold

**Pull request merged**
